src/apispec/core.py

```diff
@@ -69,6 +73,17 @@ def get_ref(obj_type, obj, openapi_major_version):
         if "responses" in operation:
             responses = OrderedDict()
             for code, response in iteritems(operation["responses"]):
+                if openapi_major_version < 3:
```

Comment:
apispec generally does not try to enforce compliance to OAS when it comes to OAS2 vs. OAS3 differences. In other words, I wouldn't have bothered raising a warning if passing 4XX when using OAS2.

I think the rationale is that it would be too much work and it would add unnecessary complexity to check everything. We only try our best to ensure the user can express what he needs, but we don't put all safeguards to ensure he doesn't mess up.

Now that you did the work, I don't have a strong feeling against keeping your warning, though. Just saying.

Reply:
OK, point taken. If you don't have a strong feeling then I won't change this, but I'd still be happy to do so if your opinion changed (or if anyone else objects the current approach). Removing the check would not cause any problems for my use-case anyway, when I wrote this code I was simply in "follow-the-spec to -a-tee" mode :)

Reply:
Yes. Let's wait for @sloria's take on the warning.

Also, if we keep the warning, I think we should still let the invalid code pass through. The user is warned but his intent is respected.

Reply:
Updated - the warning remains but the ```continue``` statement has gone, so the user's status code will always be documented.